src/main/java/org/opentripplanner/graph_builder/module/osm/GermanyWayPropertySetSource.java

```diff
@@ -0,0 +1,119 @@
+                33.33f
+        ); // = 120kph. Varies between 80 - 120 kph depending on road and season.
+        props.setCarSpeed("highway=motorway_link", 15); // = 54kph
+        props.setCarSpeed("highway=trunk", 27.27f); // 100kph
+        props.setCarSpeed("highway=trunk_link", 15); // = 54kph
+        props.setCarSpeed("highway=primary", 27.27f); // 100kph
+        props.setCarSpeed("highway=primary_link", 15); // = 54kph
+
+        // you should only use parking aisle if there is no other options
+        // ideally they would be set to noThruTraffic but that would mean the parking lots are inaccessible
+        props.setCarSpeed("service=parking_aisle", 5);
+
+        /*
+         * Many agricultural ways are tagged as 'track' but have no access tags. We assume this to mean that cars
+         * are prohibited.
+         * https://www.openstreetmap.org/way/124263424
+         */
+        props.setProperties(
+                "highway=track", StreetTraversalPermission.PEDESTRIAN_AND_BICYCLE, 1.0, 1.0);
+        props.setProperties(
+                "highway=track;surface=*", StreetTraversalPermission.PEDESTRIAN_AND_BICYCLE, 1.0,
+                1.0
+        );
+
+        props.setProperties(
+                "highway=residential;junction=roundabout", StreetTraversalPermission.ALL, 0.98,
+                0.98
+        );
+        props.setProperties(
+                "highway=*;junction=roundabout", StreetTraversalPermission.BICYCLE_AND_CAR);
+
+        // Pedestrian zones in Germany are forbidden for bicycles by default
+        props.setProperties("highway=pedestrian", StreetTraversalPermission.PEDESTRIAN);
+        props.setProperties(
+                "highway=residential;maxspeed=30", StreetTraversalPermission.ALL, 0.9, 0.9);
+        props.setProperties("highway=footway;bicycle=yes",
+                StreetTraversalPermission.PEDESTRIAN_AND_BICYCLE, 0.8, 0.8
+        );
+        // Default was 2.5, we want to favor using mixed footways somewhat
+        props.setProperties("footway=sidewalk;highway=footway;bicycle=yes",
+                StreetTraversalPermission.PEDESTRIAN_AND_BICYCLE, 1.2, 1.2
+        );
+
+        props.setProperties("highway=tertiary", StreetTraversalPermission.ALL, 1.2, 1.2, true);
+        props.setProperties("maxspeed=70", StreetTraversalPermission.ALL, 1.5, 1.5, true);
+        props.setProperties("maxspeed=80", StreetTraversalPermission.ALL, 2.0, 2.0, true);
+        props.setProperties("maxspeed=90", StreetTraversalPermission.ALL, 3.0, 3.0, true);
+        props.setProperties("maxspeed=100", StreetTraversalPermission.ALL, 5.0, 5.0, true);
+
+        /* tracktype */
+        props.setProperties(
+                "tracktype=grade1", StreetTraversalPermission.ALL, 1.0, 1.0, true); // Solid
+        props.setProperties(
+                "tracktype=grade2", StreetTraversalPermission.ALL, 1.1, 1.1,
+                true
+        ); // Solid but unpaved.
+        props.setProperties(
```

Comment:
Line comments should be before the statment, never use line comments in the right margin. Above is a good example why. It was not obvious to me first if the `// Solid but unpaved.` was commenting on the line before or after.